include/linker/devicetree_regions.h

```diff
@@ -50,30 +88,66 @@
 /**
  * @brief Declare a memory region
  *
+ * Example devicetree fragment:
+ *
+ * @code{.dts}
+ *    test_sram: sram@20010000 {
+ *        compatible = "zephyr,memory-region", "mmio-sram";
+ *        reg = < 0x20010000 0x1000 >;
+ *        zephyr,memory-region = "FOOBAR";
+ *    };
+ * @endcode
+ *
+ * will result in:
+ *
+ *    FOOBAR (rw) : ORIGIN = (0x20010000), LENGTH = (0x1000)
+ *
  * @param node_id devicetree node identifier
  * @param attr region attributes
  */
-#define _REGION_DECLARE(node_id)	      \
-	LINKER_DT_NODE_REGION_NAME(node_id) : \
-	ORIGIN = DT_REG_ADDR(node_id),	      \
+#define _REGION_DECLARE(node_id)			\
+	LINKER_DT_NODE_REGION_NAME_TOKEN(node_id) :	\
+	ORIGIN = DT_REG_ADDR(node_id),			\
 	LENGTH = DT_REG_SIZE(node_id)
 
 /**
  * @brief Declare a memory section from the device tree nodes with
  *	  compatible 'zephyr,memory-region'
  *
+ * Example devicetree fragment:
+ *
+ * @code{.dts}
+ *    test_sram: sram@20010000 {
+ *        compatible = "zephyr,memory-region", "mmio-sram";
+ *        reg = < 0x20010000 0x1000 >;
+ *        zephyr,memory-region = "FOOBAR";
+ *    };
+ * @endcode
+ *
+ * will result in:
+ *
+ *    FOOBAR 0x20010000 (NOLOAD) :
+ *    {
+ *        __FOOBAR_start = .;
+ *        KEEP(*(FOOBAR))
+ *        KEEP(*(FOOBAR.*))
+ *        __FOOBAR_end = .;
+ *    } > FOOBAR
+ *    __FOOBAR_size = __FOOBAR_end - __FOOBAR_start;
+ *    __FOOBAR_load_start = LOADADDR(FOOBAR);
```

Comment:
```suggestion
 * @code{.unparsed}
 *    FOOBAR 0x20010000 (NOLOAD) :
 *    {
 *        __FOOBAR_start = .;
 *        KEEP(*(FOOBAR))
 *        KEEP(*(FOOBAR.*))
 *        __FOOBAR_end = .;
 *    } > FOOBAR
 *    __FOOBAR_size = __FOOBAR_end - __FOOBAR_start;
 *    __FOOBAR_load_start = LOADADDR(FOOBAR);
 * @endcode
```

(will keep rendering an uncolorized code box)

Reply:
:(